caikit/runtime/service_factory.py

```diff
@@ -141,18 +142,17 @@ def get_service_package(cls, service_type: ServiceType) -> ServicePackage:
 
         rpc_list = [rpc for rpc in rpc_list if rpc.return_type is not None]
 
-        request_data_models = [
-            rpc.create_request_data_model(package_name) for rpc in rpc_list
-        ]
+        for rpc in rpc_list:
+            rpc.create_request_data_model(package_name)
 
         client_module = ModuleType(
             "ClientMessages",
             "Package with service message class implementations",
         )
 
-        for dm_class in request_data_models:
-            # We need the message class that data model serializes to
-            setattr(client_module, dm_class.__name__, type(dm_class().to_proto()))
+        for proto_class in _AUTO_GEN_PROTO_CLASSES:
+            # We need all the DM objects in the client_module for ease of use
+            setattr(client_module, proto_class.DESCRIPTOR.name, proto_class)
```

Comment:
After hacking a bit with @sboagibm today, I think we'll end up wanting something even more robust here.

The `messages` module can only be used if you happen to know the name of the thing you're requesting, which isn't super useful in many contexts. We wanted to write a generic initializer that would spin up a shim module to another caikit runtime, but then we had to write things like:
```
request_class = getattr(service_package.messages, f"{task_class}Request")
```

Food for thought: what if we had something like this here? And returned full data models instead of just the proto messages?
```
def get_inference_request(module_class: Type[ModuleBase]) -> Type[DataBase]:

def get_training_request(module_class: Type[ModuleBase]) -> Type[DataBase]:

def get_training_parameters(module_class: Type[ModuleBase]) -> Type[DataBase]:
```

Reply:
(food for _future_ thought, to be clear)